### PR TITLE
fixed bug with removing elements from list while iterating on it

### PR DESCRIPTION
**Pull request by d1skort**
Hi!

I've found this interesting bug and want to help you fix it! :)

### CODE REVIEW

**Comment by d1skort** [comment]
Any chance to merge?

**Comment by srinivasreddy** [comment]
Ping @vmalloc  @mdomke

File: mongomock/collection.py [hunk 1]
```diff
@@ -506,14 +506,15 @@ def _update(self, spec, document, upsert=False, manipulate=False,
                             if not isinstance(arr, list):
                                 continue
 
+                            arr_copy = copy.deepcopy(arr)
                             if isinstance(value, dict):
-                                for idx, obj in enumerate(arr):
+                                for obj in arr_copy:
                                     if filter_applies(value, obj):
-                                        del arr[idx]
+                                        arr.remove(obj)
```
Comment (vmalloc) [inline mongomock/collection.py]: This seems wrong... Is it always guaranteed to be the first? Remove only removes the first match in the array...

**Comment by vmalloc** [comment]
Ok. Thanks!

**Pull request merged**